core/src/main/java/org/apache/carbondata/core/dictionary/client/DictionaryClientHandler.java

```diff
@@ -77,20 +79,13 @@ public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws
    */
   public DictionaryKey getDictionary(DictionaryKey key) {
     DictionaryKey dictionaryKey;
-    BlockingQueue<DictionaryKey> dictKeyQueue = null;
     try {
-      synchronized (lock) {
-        dictKeyQueue = dictKeyQueueMap.get(key.getThreadNo());
-        if (dictKeyQueue == null) {
-          dictKeyQueue = new LinkedBlockingQueue<DictionaryKey>();
-          dictKeyQueueMap.put(key.getThreadNo(), dictKeyQueue);
-        }
-      }
-      byte[] serialize = KryoRegister.serialize(key);
-      ctx.getChannel().write(serialize);
+      ByteBuf buffer = ctx.alloc().buffer();
+      key.writeData(buffer);
+      ctx.writeAndFlush(buffer);
     } catch (Exception e) {
-      LOGGER.error("Error while send request to server " + e.getMessage());
-      ctx.getChannel().close();
+      LOGGER.error(e, "Error while send request to server " + e.getMessage());
+      ctx.close();
     }
     boolean interrupted = false;
     try {
```

Comment:
I am not sure about SettingFuture, but i don't see difference between blocking Queue and SettingFuture.